src/compas/geometry/_shapes/sphere.py

```diff
@@ -289,9 +292,9 @@ def transform(self, transformation):
         >>> sphere.transform(T)
 
         """
-        Sc, Sh, R, Tl, P = transformation.decomposed()
-        self.point.transform(Tl)
-        self.radius *= max(Sc.scale_factors)
+        self.point.transform(transformation)
+        Sc, _, _, _, _ = transformation.decomposed()
```

Comment:
this is an expensive operation (decomposing the transformation matrix in independent components). scale and translation are completely independent. is this absolutely necessary?

Reply:
Do you have a suggestion?

Reply:
well if the matrix only contains scale and translation, you could just do
```python
factors = [transformation[0, 0], transformation[1, 1], transformation[2, 2]
vector = [transformation[0, 3], transformation[1, 3], transformation[2, 3]
```
no, @romanarust ?

Reply:
But we cannot ensure the matrix only contains scale and translation.
Additionally, `transform` does imply a potentially costly operation, so, I would err on the side of correctness rather than optimize for speed while leaving room for potentially incorrect results.

Reply:
okay cool